config/placeholder_replacement.go

```diff
@@ -0,0 +1,136 @@
+package config
+
+import (
+	"reflect"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"go.viam.com/rdk/utils"
+)
+
+// Example strings that satisfy this regexp:
+// ${packages.my-COOL-ml-model/__89}
+// ${packages.modules.intel:CameraThatRocks}
+var packagePlaceholderRegexp = regexp.MustCompile(`\$\{(?P<fulltriple>packages(\.(?P<type>ml_models|modules))?\.(?P<name>[\w:\/-]+))\}`)
```

Comment:
Only major difference is that this regex does not require the replacement to be the first character of the string. 

I don't know if we want this... The only reason I removed it was because I was testing the replacement with my module and thought there was some bug because my string wasn't being replaced... whatever other people think